### PR TITLE
Delete encrypted storage in a try catch to avoid errors.

### PR DESCRIPTION
**Pull request by DavidKutu**
For #5934

-   [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [x] Title summarizes what is changing.
-   [x] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

File: src/client/common/application/encryptedStorage.ts [hunk 1]
```diff
@@ -40,8 +45,12 @@ export class EncryptedStorage implements IEncryptedStorage {
             return val;
         } catch (e) {
             // If we get an error trying to get a secret, it might be corrupted. So we delete it.
-            await this.extensionContext.secrets.delete(`${service}.${key}`);
-            return;
+            try {
+                this.extensionContext.secrets.delete(`${service}.${key}`);
```
Comment (rchiodo) [inline src/client/common/application/encryptedStorage.ts:49]: I believe this won't be caught here. The promise is ignored.

You might try this instead:
```
this.extensionContext.secrets.delete(`${service}.${key}`).catch(traceError);
```

**Review by rchiodo — CHANGES_REQUESTED** [review]
⏲️

File: src/client/common/application/encryptedStorage.ts (hunk 1 above)
Comment (DavidKutu) [reply src/client/common/application/encryptedStorage.ts:49]: thenables don't have .catch

Comment (DavidKutu) [reply src/client/common/application/encryptedStorage.ts:49]: I'll put the await back

**Review by rchiodo — APPROVED** [review]

**Review by IanMatthewHuff — APPROVED** [review]

**Pull request merged**